parsers/MD.py

```diff
@@ -1,103 +1,96 @@
-# Data for the fuel-mix at CERS Moldovenească for the year 2020 can be found at:
-# http://moldgres.com/wp-content/uploads/2021/02/proizvodstvennye-pokazateli-zao-moldavskaja-gres-za-2020-god.pdf
-
-# Annual reports from moldelectrica can be found at:
-# https://moldelectrica.md/ro/network/annual_report
-
-
-def template_price_response(zone_key: str, datetime: datetime, price) -> dict:
-    return {
-        "zoneKey": zone_key,
-        "datetime": datetime,
-        "currency": "MDL",
-        "price": price,
-        "source": "moldelectrica.md",
-    }
-
-
-def template_consumption_response(
-    zone_key: str, datetime: datetime, consumption
-) -> dict:
-    return {
-        "zoneKey": zone_key,
-        "datetime": datetime,
-        "consumption": consumption,
-        "source": "moldelectrica.md",
-    }
-
-
-def template_production_response(zone_key: str, datetime: datetime, production) -> dict:
-    return {
-        "zoneKey": zone_key,
-        "datetime": datetime,
-        "production": production,
-        "storage": {},
-        "source": "moldelectrica.md",
-    }
-
-
-def template_exchange_response(
-    sorted_zone_keys: str, datetime: datetime, netflow
-) -> dict:
-    return {
-        "sortedZoneKeys": sorted_zone_keys,
-        "datetime": datetime,
-        "netFlow": netflow,
-        "source": "moldelectrica.md",
-    }
-
-
-def get_archive_data(
+ARCHIVE_BASE_URL = "https://moldelectrica.md/utils/archive2.php?id=table&type=array"
+SOURCE = "moldelectrica.md"
+
+
+# Moldoelectrica electricity tariffs as defined by government-agency decisions.
+_MOLDOELECTRICA_NEW_POWER_PRICE_IN_MDL_PER_MW = {
+    # https://www.legis.md/cautare/getResults?doc_id=78826&lang=ro
+    datetime(2000, 4, 1, tzinfo=timezone.utc): 18.8,
+    # https://www.legis.md/cautare/getResults?doc_id=103953&lang=ro
+    datetime(2001, 10, 1, tzinfo=timezone.utc): 28.0,
+    # https://www.legis.md/cautare/getResults?doc_id=40249&lang=ro
+    datetime(2002, 9, 1, tzinfo=timezone.utc): 35.2,
+    # https://www.legis.md/cautare/getResults?doc_id=42701&lang=ro
+    datetime(2005, 9, 1, tzinfo=timezone.utc): 39.3,
+    # https://www.legis.md/cautare/getResults?doc_id=10948&lang=ro
+    datetime(2007, 8, 3, tzinfo=timezone.utc): 51.8,
+    # https://www.legis.md/cautare/getResults?doc_id=40130&lang=ro
+    datetime(2010, 1, 19, tzinfo=timezone.utc): 63.2,
+    # https://www.legis.md/cautare/getResults?doc_id=40589&lang=ro
+    datetime(2012, 5, 11, tzinfo=timezone.utc): 80.2,
+    # https://www.legis.md/cautare/getResults?doc_id=84436&lang=ro
+    datetime(2015, 7, 31, tzinfo=timezone.utc): 145.0,
+    # https://www.legis.md/cautare/getResults?doc_id=134854&lang=ro
+    datetime(2023, 12, 31, tzinfo=timezone.utc): 201.0,
+    # https://www.legis.md/cautare/getResults?doc_id=142391&lang=ro
+    datetime(2024, 3, 21, tzinfo=timezone.utc): 185.0,
+    # TODO(amv213): update as new tariffs get rolled out...
+}
+
+
+class ArchiveDatapoint(NamedTuple):
```

Comment:
Should this have a `@dataclass` decorator?

Reply:
We can turn it into a dataclass instead of a named tuple if we prefer, but otherwise a `NamedTuple` doesn't need that decorator. 

I quickly updated the existing `collections.namedtuple` to a `typing.NamedTuple` - because it allows defining them in this more dataclass-y style that I find much more readable / intuitive

Reply:
It's fine, I don't really have a preference either way. :)